source/settings.cpp

```diff
@@ -1088,6 +1111,87 @@ namespace Settings {
     ResolveExcludedLocationConflicts();
   }
 
+  // Randomizes all settings in a category if chosen
+  // Hides all relevant options
+  void RandomizeAllSettings() {
+    // Open Settings
+    if (RandomizeOpen) {
+      // Skip Logic and RandomizeOpen Options to ensure proper logic
+      for (u8 i=2; i < openOptions.size(); i++) {
+        //hide options
+        openOptions[i]->Hide();
+
+        //randomize options
+        openOptions[i]->SetSelectedIndex(rand() % openOptions[i]->GetOptionCount());
+      }
+      // Randomize Ganon Trials
+      RandomGanonsTrials.SetSelectedIndex(ON);
+    }
+    else {
+      for (u8 i=2; i < openOptions.size(); i++) {
```

Comment:
```suggestion
      for (size_t i = 2; i < openOptions.size(); i++) {
```

Ditto for other loops below